src/som/interpreter/actors/ReceivedRootNode.java

```diff
@@ -47,6 +52,22 @@ protected final void resolvePromise(final VirtualFrame frame,
     resolve.executeEvaluated(frame, resolver, result, isBreakpointOnPromiseResolution);
   }
 
+  protected final void ruinPromise(final VirtualFrame frame,
+      final SResolver resolver, final SAbstractObject exception) {
+    // lazy initialization of resolution node
+    if (ruin == null) {
+      CompilerDirectives.transferToInterpreterAndInvalidate();
+      if (resolver == null) {
+        throw new SomException(exception); //I'm ruining a promise without a resolver. No onError can be added so rethrow the exception
```

Comment:
This is wrong behavior, this will crash the thread.
If the user decided not to capture the result, we can't crash the VM. This is part of the language semantics. A `NullResolver` should handle this case, like in the normal resolution case.